kore/src/Kore/Log/DebugRewriteTrace.hs

```diff
@@ -0,0 +1,267 @@
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+{- |
+Copyright   : (c) Runtime Verification, 2022
+License     : BSD-3-Clause
+-}
+module Kore.Log.DebugRewriteTrace (
+    DebugInitialClaim (..),
+    DebugInitialPattern (..),
+    DebugFinalPatterns (..),
+    DebugRewriteTrace (..),
+    debugInitialClaim,
+    debugInitialPattern,
+    debugFinalPatterns,
+    debugRewriteTrace,
+    rewriteTraceLogger,
+) where
+
+import Data.ByteString (
+    ByteString,
+ )
+import Data.Text (
+    Text,
+ )
+import Data.Text.Encoding (
+    decodeUtf8,
+ )
+import Data.Yaml (
+    ToJSON,
+    Value (..),
+    array,
+    encode,
+    object,
+    toJSON,
+    (.=),
+ )
```

Comment:
I think we should not introduce a dependency on `yaml`. We currently depend on `conduit-extra` because of `json-rpc` but I don't see the need to use the yaml package here.
The logger prints out a newline-separated sequence of (yaml) objects and  any valid json is valid yaml , too. We can maybe just as well use `aeson-pretty` instead and produce valid yaml without any other changes.
(the reason that this is yaml and not json is that the top-level sequence is not enclosed in `[` and `]` and separated by newlines instead of commas).

Reply:
I don't feel strongly about this either way. @ana-pantilie ?

Reply:
Let's keep it for now, I wouldn't spend more time on this if it's functional and prints the output as desired.